protocol/xmrmaker/net.go

```diff
@@ -14,33 +14,39 @@ func (b *Instance) Provides() types.ProvidesCoin {
 	return types.ProvidesXMR
 }
 
-func (b *Instance) initiate(offer *types.Offer, offerExtra *types.OfferExtra, providesAmount common.MoneroAmount,
-	desiredAmount common.EtherAmount) error {
+func (b *Instance) initiate(
+	offer *types.Offer,
+	offerExtra *types.OfferExtra,
+	providesAmount common.MoneroAmount,
+	desiredAmount common.EtherAmount,
+) (*swapState, error) {
```

Comment:
This function now returns the swapState so the caller doesn't have to grab the lock again to know what we changed it to.